autotest/pyscripts/test_gdal2tiles.py

```diff
@@ -101,10 +101,38 @@ def test_gdal2tiles_py_cleanup():
 
     return 'success'
 
+
+def test_does_not_error_when_source_bounds_close_to_tiles_bound():
+    """
+    Case where the geo coords of the image top Y point gives a corresponding tile of X, Y, Z
+    BUT where only a portion of the pixels of the first pixel line would be on this tile.
+    When computing the raster for the tile, we don't get any data (the pixel line is "assigned" to
+    the tile below)
+    """
+    out_folder = 'tmp/out_gdal2tiles_bounds_approx'
+    try:
+        shutil.rmtree(out_folder)
+    except:
+        pass
+
+    script_path = test_py_scripts.get_py_script('gdal2tiles')
+    if script_path is None:
+        return 'skip'
+
+    # Will blow an exception if the case is not handled properly
+    test_py_scripts.run_py_script(
```

Comment:
That does not work because the execution of gdal is wrapped like that
```python
def run_func(func):
    try:
        result = func()
        print(result)
        return result
    except SystemExit as x:
        import traceback
        traceback.print_exc()

        raise x
    except:
        result = 'fail (blowup)'
        print(result)

        import traceback
        traceback.print_exc()
        return result
```

I find the test runner really complex and abstracting a lot of features (too many) of a normal test runner.

Anyway, I can either leave it like that or I can try and call gdal2tiles myself, doing the import myself, rather than going through the `test_py_script` abstraction.
What do you reckon is best?